test/expected/timestamp.out

```diff
@@ -157,15 +157,24 @@ SELECT _timescaledb_internal.to_unix_microseconds('2017-02-07 15:09:36.236538+00
      1486480176236538
 (1 row)
 
--- In UNIX microseconds, BIGINT MAX is smaller than internal date upper bound
--- and should therefore be OK. Further, converting to the internal postgres
--- epoch cannot overflow a 64-bit INTEGER since the postgres epoch is at a
--- later date compared to the UNIX epoch, and is therefore represented by a
--- smaller number
+-- For timestamps, BIGINT MAX represents +Infinity. We keep this
+-- notion for UNIX epoch.
 SELECT _timescaledb_internal.to_timestamp(9223372036854775807);
+ to_timestamp 
+--------------
+ infinity
+(1 row)
+
+-- In UNIX microseconds, the largest bigint value below infinity
+-- (BIGINT MAX) is smaller than internal date upper bound and should
+-- therefore be OK. Further, converting to the internal postgres epoch
+-- cannot overflow a 64-bit INTEGER since the postgres epoch is at a
+-- later date compared to the UNIX epoch, and is therefore represented
+-- by a smaller number
+SELECT _timescaledb_internal.to_timestamp(9223372036854775806);
              to_timestamp              
 ---------------------------------------
- Sun Jan 10 04:00:54.775807 294247 UTC
+ Sun Jan 10 04:00:54.775806 294247 UTC
```

Comment:
This is an odd case and I would say the previous behavior is a bug. `9223372036854775806` is normally `+Infinity` and not a "regular" timestamp value (i.e., one that translates to a date). It becomes a date because we convert it to the PG epoch.

Not sure it needs to be in the changelog, except maybe as a bug. I'd rather actually try to change how we represent time internally to use native PG formats. That would resolve all these weird time conversion issues.

Reply:
Sorry, to clarify: the input here is a UNIX epoch timestamp, and to be able to convert a timestamp which is "Infinity" to UNIX epoch, we need to have a representation of "Infinity" in the UNIX epoch. In other words, we reserve PG_INT64_MAX for this, just like timestamp. I think that we didn't have this before was a bug, and we could potentially report it as such in the changelog.